### PR TITLE
add a ACEReaderWithTrueCaseFixer class

### PR DESCRIPTION
**Pull request by Slash0BZ**
update usages upon this new class
Also a bug fix that cases opposite direction relations in RelationAnnotator.
Also added two new testing functions in MD and RE.

### CODE REVIEW

**Comment by Slash0BZ** [comment]
closes #580

**Comment by danyaljj** [comment]
I was thinking that it might be a good idea if we move `ACEReaderWithTrueCaseFixer.java` (which is in corpus readers) to `external/stanford_3.3.1` (hence we don't have to add stanford as a dependency to corpus-readers). What do you think? (@mssammon feel free to comment on this)

**Comment by Slash0BZ** [comment]
Ah that way ACEReaderWithTrueCaseFixer will be a package in stanford_3.3.1 so it can't extend ACEReader anymore right?

**Comment by mssammon** [comment]
the new reader can be in the external/ package, I agree... maybe point to it in the corpusreader documentation, though.  I don't see why the reader can't extend ACEReader; package name/path doesn't have to be identical.

**Comment by danyaljj** [comment]
So rather than copus-readers being dependant on stanford, stanford would be dependant on stanford.

**Comment by Slash0BZ** [comment]
Just making sure, are we moving it to `edu.illinois.cs.cogcomp.pipeline.common` which holds all the external sources?

**Comment by danyaljj** [comment]
No. Inside `stanford_3.3.1`: 
https://github.com/CogComp/cogcomp-nlp/tree/master/external/stanford_3.3.1

**Comment by Slash0BZ** [comment]
I understand, but all files in `stanford_3.3.1` are in package `edu.illinois.cs.cogcomp.pipeline.common`.

**Comment by danyaljj** [comment]
Ah you can keep the package naming compatible with corpus reader (which is what you currently have).

File: external/stanford_3.3.1/pom.xml [hunk 1]
```diff
@@ -33,5 +33,10 @@
             <version>3.3.1</version>
             <classifier>models</classifier>
         </dependency>
+        <dependency>
+            <groupId>edu.illinois.cs.cogcomp</groupId>
+            <artifactId>illinois-corpusreaders</artifactId>
+            <version>3.1.35</version>
```
Comment (danyaljj) [inline external/stanford_3.3.1/pom.xml]: 3.1.36

**Comment by danyaljj** [comment]
Minor comment on versioning: merge after fixing it (and CI turning green).

**Pull request merged**